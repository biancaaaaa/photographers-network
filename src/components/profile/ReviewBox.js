import React from "react";
import { Avatar } from "../label/Avatar";
import { ArrowSVG } from "../svg/ArrowSVG";
import PropTypes from "prop-types";

export const ReviewBox = ({ title, userImageURL, name, quote }) => (
  <div className="review-box">
    <h2 className="review-box-title">{title}</h2>
    <Avatar userImageUrl={userImageURL} />
    <h3 className="review-author-name">{name}</h3>
    <p className="review-author-quote">“{quote}”</p>
    <ArrowSVG classes="gb-icon-medium" />
  </div>
);

ReviewBox.propTypes = {
  title: PropTypes.string.isRequired,
  userImageURL: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  quote: PropTypes.string.isRequired
};
