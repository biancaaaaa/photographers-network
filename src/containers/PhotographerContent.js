import React, {Component} from 'react';
import {Link} from 'react-router-dom';

class PhotographerContent extends Component {
    render() {
        return (
            <div className="section-content normalized">
                <h3>Job Offers</h3>
                <Link to="/" className="gb-btn gb-btn-medium gb-btn-primary">See all job offers...</Link>
                <h3>My Jobs</h3>
                <p>Here are my jobs</p>
                <Link to="/my-jobs" className="gb-btn gb-btn-medium gb-btn-primary">View all of my jobs</Link>
            </div>
        )
    }
}

export default PhotographerContent;