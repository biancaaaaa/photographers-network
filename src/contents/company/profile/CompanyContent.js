import React from "react";
import { Link } from "react-router-dom";

const CompanyContent = ({ isOtherUser }) => {
  if (!isOtherUser) {
    return (
      <div className="section-content normalized">
        <Link to="../createJob">Create job offer</Link>
        <p>Here you can create your job offer</p>
        <h3>Search photographers</h3>
        <Link
          to="/search-photographers"
          className="gb-btn gb-btn-medium gb-btn-primary"
        >
          Search now...
        </Link>
      </div>
    );
  }
  return <h2>Not your profile</h2>;
};
export default CompanyContent;
