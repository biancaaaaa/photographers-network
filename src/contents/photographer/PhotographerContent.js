import React from "react";
import { Link } from "react-router-dom";
import { PortofolioGallery } from "../../components/PortofolioGallery";
import { PropTypes } from "prop-types";

export const PhotographerContent = ({ user }) => (
  <React.Fragment>
    {user.portofolio.length > 0 && (
      <PortofolioGallery photosList={user.portofolio} />
    )}
    <div className="section-content normalized">
      <h3>Job Offers</h3>
      <Link to="/" className="gb-btn gb-btn-medium gb-btn-primary">
        See all job offers...
      </Link>
      <h3>My Jobs</h3>
      <p>Here are my jobs</p>
      <Link to="/my-jobs" className="gb-btn gb-btn-medium gb-btn-primary">
        View all of my jobs
      </Link>
    </div>
  </React.Fragment>
);

PhotographerContent.propTypes = {
  user: PropTypes.object.isRequired
};
