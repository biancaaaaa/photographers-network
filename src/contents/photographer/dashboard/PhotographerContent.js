import React from "react";
import { Link } from "react-router-dom";
import { PortofolioGallery } from "../../../components/PortofolioGallery";
import { PropTypes } from "prop-types";

export const PhotographerContent = ({ photographerData, isOtherUser }) => (
  <React.Fragment>
    {photographerData.portofolio && photographerData.portofolio.length > 0 && (
      <PortofolioGallery photosList={photographerData.portofolio} />
    )}
    {!isOtherUser && <OwnPhotographerProfile />}
  </React.Fragment>
);

const OwnPhotographerProfile = () => (
  <div className="section-content normalized">
    <h3>Job Offers</h3>
    <Link to="/" className="gb-btn gb-btn-medium gb-btn-primary">
      See all job offers...
    </Link>
    <h3>My Jobs</h3>
    <p>Here are my jobs</p>
    <Link to="/my-jobs" className="gb-btn gb-btn-medium gb-btn-primary">
      View all of my jobs
    </Link>
  </div>
);

PhotographerContent.propTypes = {
  photographerData: PropTypes.object.isRequired
};
