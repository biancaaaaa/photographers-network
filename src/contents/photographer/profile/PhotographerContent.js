import React from "react";
import {PortofolioGallery} from "../../../components/PortofolioGallery";
import {PropTypes} from "prop-types";
import {HireButton} from "../../../components/HireButton";
import {Link} from "react-router-dom";
import {ReviewBox} from "../../../components/profile/ReviewBox";

export const PhotographerContent = ({reviews, photographerData, isOtherUser, siggnedInUser: signedInUser, uid, finishedJobs, hireable}) => (
  <React.Fragment>
    <div className="flex-container-space">
      <div>
        <span className="big-num">{finishedJobs}</span>
        <span className="light-blue">Jobs <br/>done</span>
      </div>
      <Link
        to={`/finished-jobs/${uid}`}
        className="gb-btn gb-btn-small gb-btn-primary gb-btn-white"
      >
        See the projects
      </Link>
    </div>
    {photographerData.portfolio && photographerData.portfolio.length > 0 && (
      <React.Fragment>
        <h2 className="black-header-portfolio">Portfolio</h2>
        <PortofolioGallery photosList={photographerData.portfolio}/>
      </React.Fragment>
    )}
    {
      (hireable && signedInUser.type === "company") &&
      <div className="pink-box">
        <p>There, between rolling hills and the Apennine mountains in the Mugello valley.</p>
        <HireButton
          classes="hire-button"
          photographerData={photographerData}
          siggnedInUser={signedInUser}
          uid={uid}
          photographerName={`${photographerData.firstName} ${photographerData.lastName}`}
        />
      </div>
    }
    {
      reviews[0] &&
      <div className="relative-container">
        <Link className="absolute-content" to={`/reviews/${uid}`}/>
        <ReviewBox title="Last review"
                   userImageURL={reviews[0].authorData.profileImageUrl}
                   name={reviews[0].authorData.companyName}
                   quote={reviews[0].message}
        />
      </div>
    }

  </React.Fragment>
);

PhotographerContent.propTypes = {
  photographerData: PropTypes.object.isRequired
};
