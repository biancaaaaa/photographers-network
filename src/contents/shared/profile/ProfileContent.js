import React from "react";
import { PhotographerContent } from "../../photographer/profile/PhotographerContent";
import { HireButton } from "../../../components/HireButton";
import { Link } from "react-router-dom";
import { PencilSVG } from "../../../components/svg/PencilSVG";
import ProfileDescription from "./ProfileDescription";
import PropTypes from "prop-types";

export const ProfileContent = ({
  reviews,
  thisProfileData,
  currentUserData,
  currentUserId,
  otherUser,
  finishedJobs
}) => (
  <div className="profile-content">
    <h1>
      {thisProfileData.companyName ||
        `${thisProfileData.firstName} ${thisProfileData.lastName}`}
    </h1>
    <p>
      <span className="profile-desc bold">
        {Object.values(thisProfileData.locations)[0].city}
      </span>
      <span> | </span>
      <span className="profile-desc user-type">{thisProfileData.type}</span>
    </p>
    {thisProfileData.type === "photographer" &&
      currentUserData.type === "company" &&
      thisProfileData.hireable && (
        <HireButton
          classes="gb-btn gb-btn-small gb-btn-primary gb-btn-white hire"
          photographerData={thisProfileData}
          siggnedInUser={currentUserData}
          uid={currentUserId}
          photographerName={`${thisProfileData.firstName} ${
            thisProfileData.lastName
          }`}
        />
      )}
    {!otherUser && (
      <Link
        to="/ProfileEdit"
        className="gb-btn gb-btn-small gb-btn-primary gb-btn-white"
      >
        Edit Profile <PencilSVG classes="gb-icon-medium gb-icon-fill-white" />
      </Link>
    )}
    <ProfileDescription
      description={thisProfileData.description}
      otherUser={otherUser}
    />

    {thisProfileData.type === "photographer" ? (
      <PhotographerContent
        photographerData={thisProfileData}
        isOtherUser={otherUser}
        siggnedInUser={currentUserData}
        uid={currentUserId}
        finishedJobs={finishedJobs}
        hireable={thisProfileData.hireable}
        reviews={reviews}
      />
    ) : (
      <div>Rest of the company content</div>
    )}
  </div>
);
