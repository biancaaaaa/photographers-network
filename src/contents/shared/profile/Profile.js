// dependencies
import React, { Component } from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import { fetchProfileInfos } from "../../../redux/actions/profile-action";

// components
import { ProfileCard } from "../../../components/ProfileCard";
import { LinkLists } from "../../../components/LinkLists";

// contents
import { PhotographerContent } from "../../photographer/dashboard/PhotographerContent";
import CompanyContent from "../../company/profile/CompanyContent";

class Profile extends Component {
  componentDidMount() {
    const { profilesData, user } = this.props;

    const profileId = this.props.match.params.uid;

    if (!profilesData[profileId]) {
      console.log("nu e");
      this.props.fetchProfile(profileId);
    }
  }

  render() {
    const { fetchingProfile, profilesData, error, match, user } = this.props;
    const queryId = match.params.uid;
    const isOtherUser = queryId !== user.uid;
    const pageLinks = [
      { txt: "Facebook", link: "www.facebook.com" },
      { txt: "Twitter", link: "www.twitter.com" }
    ];

    let thisProfileData = profilesData[queryId];

    if (fetchingProfile) {
      return <h2>Fetching Profile Data...</h2>;
    }

    if (error) {
      return <h2>404 ERROR</h2>;
    }

    return (
      <ProfileView
        isOtherUser={isOtherUser}
        thisProfileData={thisProfileData}
        pageLinks={pageLinks}
        siggnedInUser={user}
      />
    );
  }
}

const mapStateToProps = state => {
  const profiles = state.profiles;
  const user = state.user;
  return {
    user: user.userData,
    userOn: user.userOn,
    profilesData: profiles.data,
    fetchingProfile: profiles.fetchingProfile,
    error: profiles.error
  };
};

const mapDispatchToProps = dispatch => ({
  fetchProfile: uid => dispatch(fetchProfileInfos(uid))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Profile);

const ProfileView = ({
  isOtherUser,
  thisProfileData,
  pageLinks,
  siggnedInUser
}) => (
  <div>
    <div className="profile">
      <ProfileCard {...thisProfileData} siggnedInUser={siggnedInUser}>
        {thisProfileData.displayName}
      </ProfileCard>
      <div className="profile-content">
        <LinkLists
          links={pageLinks}
          txtClasses="gb-text-black-opacity-30 gb-subtitle-medium"
          liClasses="footer-nav-item"
        />
        {!isOtherUser && (
          <Link
            to="/ProfileEdit"
            className="gb-btn gb-btn-medium gb-btn-primary"
          >
            Edit Profile
          </Link>
        )}
      </div>

      {thisProfileData.type === "photographer" ? (
        <PhotographerContent
          photographerData={thisProfileData}
          isOtherUser={isOtherUser}
        />
      ) : (
        <CompanyContent isOtherUser={isOtherUser} />
      )}
    </div>
  </div>
);
