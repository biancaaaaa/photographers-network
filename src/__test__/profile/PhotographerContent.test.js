import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";

import makeUser from "../mockUser";
import { shallow } from "enzyme";
import { PhotographerContent } from "../../contents/photographer/profile/PhotographerContent";

describe("PhotographerContent", () => {
  const photographerData = makeUser({ type: "photographer" });
  const siggnedInUser = makeUser({ type: "company" });

  it("matches the snapshot", () => {
    const tree = renderer
      .create(
        <Router>
          <PhotographerContent
            reviews={[]}
            photographerData={photographerData}
            siggnedInUser={siggnedInUser}
            uid={photographerData.uid}
            finishedJobs={photographerData.finishedJobs}
            hireable={photographerData.hireable}
          />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  const shallowElement = shallow(
    <PhotographerContent
      reviews={[]}
      photographerData={photographerData}
      siggnedInUser={siggnedInUser}
      uid={photographerData.uid}
      finishedJobs={photographerData.finishedJobs}
      hireable={photographerData.hireable}
    />
  );

  it("renders the correct number of finishedJobs", () => {
    expect(shallowElement.find(".flex-container-space .big-num").text()).toBe(
      "0"
    );

    shallowElement.setProps({ finishedJobs: 10 });

    expect(shallowElement.find(".flex-container-space .big-num").text()).toBe(
      "10"
    );
  });

  describe("renders the portofolio correctly, ", () => {
    it("if the photographer have no portfolio data", () => {
      const newPhotographerData = makeUser({
        type: "photographer",
        portfolio: null
      });
      shallowElement.setProps({ photographerData: newPhotographerData });

      expect(shallowElement.find("PortofolioGallery").length).toBe(0);
    });

    it("shows it when photographer have a portfolio", () => {
      const newPhotographerData = makeUser({
        type: "photographer",
        portfolio: []
      });
      shallowElement.setProps({ photographerData: newPhotographerData });

      expect(shallowElement.find("PortofolioGallery").length).toBe(0);
    });

    it("shows nothing if the portfolio is empty", () => {
      const newPhotographerData = makeUser({
        type: "photographer",
        portfolio: ["img1.jpg", "img2.jpg"]
      });
      shallowElement.setProps({ photographerData: newPhotographerData });

      expect(shallowElement.find("PortofolioGallery").length).toBe(1);
    });
  });

  describe("renders the hire button correctly", () => {
    it("shows it , if photographer is hireable and the currentUser is a company", () => {
      const siggnedInUser = makeUser({ type: "company" });
      const photographer = makeUser({ type: "photographer", hireable: true });

      shallowElement.setProps({
        photographerData: photographer,
        siggnedInUser: siggnedInUser,
        hireable: true
      });

      expect(shallowElement.find("HireButton").length).toBe(1);
    });

    it("does not show it , if the photographer is not hireable", () => {
      const siggnedInUser = makeUser({ type: "company" });
      const photographer = makeUser({ type: "photographer", hireable: false });

      shallowElement.setProps({
        photographerData: photographer,
        siggnedInUser: siggnedInUser,
        hireable: false
      });

      expect(shallowElement.find("HireButton").length).toBe(0);
    });

    it("does not show it , if the currentUser is not a company", () => {
      const siggnedInUser = makeUser({ type: "photographer" });
      const photographer = makeUser({ type: "photographer", hireable: true });

      shallowElement.setProps({
        photographerData: photographer,
        siggnedInUser: siggnedInUser,
        hireable: true
      });

      expect(shallowElement.find("HireButton").length).toBe(0);
    });
  });

  describe("Review box", () => {
    it("does not render it , if ther are no reviews", () => {
      expect(shallowElement.find("ReviewBox").exists()).toBe(false);
    });

    it("does render it , if ther are reviews", () => {
      const shallowElementWithReviews = shallow(
        <PhotographerContent
          reviews={[
            {
              title: "Last review",
              authorData: {
                profileImageUrl: "j.jpg",
                companyName: "Test",
                quote: "Test"
              }
            }
          ]}
          photographerData={photographerData}
          siggnedInUser={siggnedInUser}
          uid={photographerData.uid}
          finishedJobs={photographerData.finishedJobs}
          hireable={photographerData.hireable}
        />
      );

      expect(shallowElementWithReviews.find("ReviewBox").exists()).toBe(true);
    });
  });
});

/*
  it("renders the link to project once", () => {
    expect(
      shallowElement.containsMatchingElement(
        <Link to="/dashboard">See the projects</Link>
      )
    ).toBe(true);
  });
*/

/*



*/
