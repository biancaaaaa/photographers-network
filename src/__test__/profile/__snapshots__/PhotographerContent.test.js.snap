// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotographerContent matches the snapshot 1`] = `
Array [
  <div
    className="flex-container-space"
  >
    <div>
      <span
        className="big-num"
      >
        0
      </span>
      <span
        className="light-blue"
      >
        Jobs 
        <br />
        done
      </span>
    </div>
    <a
      className="gb-btn gb-btn-small gb-btn-primary gb-btn-white"
      href="/finished-jobs/zzz"
      onClick={[Function]}
    >
      See the projects
    </a>
  </div>,
  <div
    className="pink-box"
  >
    <p>
      There, between rolling hills and the Apennine mountains in the Mugello valley.
    </p>
    <div
      onClick={[Function]}
    >
      <div
        className="hire-button"
      >
        <h5>
          Hire me
        </h5>
      </div>
    </div>
  </div>,
]
`;
